### ISSUE TITLE
Create Label class in ppc64 assembler.

### ISSUE THREAD

**Issue by racardoso** [PPC64/hhvm#8]
Need to create a Label class to translate (Label/Vlabel) data structure into code block address. This data structure are used in jumps and calls.
- [x] Implement `Label` Class based on X64.
- [x] Define and implement `BranchConditions` for a better interface to conditional branches.

**Comment by gut** [comment]
Implemented with bugfixes on 629a4625f7b1c0a05122f4aaf6f7515ae066c103

**Comment by gut** [comment]
Testing in `hphp/runtime/vm/jit/unique-stubs-ppc64.cpp`. 

Input code:

```
ppc64_asm::Label test, test2;
Reg64 arg1 = argNumToRegName[0];
Reg64 arg2 = argNumToRegName[1];

a.li(arg1, 1);
a.li(arg2, 2);

asm_label     (a, test);

a.cmpd(arg1,arg2);
a.bc          (test, ppc64_asm::BranchConditions::LessThan         );
a.bc          (test, ppc64_asm::BranchConditions::LessThanEqual    );
a.bc          (test, ppc64_asm::BranchConditions::GreaterThan      );
a.bc          (test, ppc64_asm::BranchConditions::GreaterThanEqual );
a.bc          (test, ppc64_asm::BranchConditions::Equal            );
a.bc          (test, ppc64_asm::BranchConditions::NotEqual         );
a.b           (test);
a.branchAuto  (test, rAsm, ppc64_asm::BranchConditions::Always, ppc64_asm::LinkReg::Save);

ppc64_asm::Label abs(CodeAddress(0x1000));
a.bca         (abs, ppc64_asm::BranchConditions::Always);

a.b           (test2);
asm_label     (a, test2);
a.bc          (test, ppc64_asm::BranchConditions::GreaterThanEqual);


ppc64_asm::BranchParams bp(ppc64_asm::BranchConditions::Always);
const ssize_t address = ssize_t(a.frontier());
Reg64 tmp = rAsm;

// Optimization: the highest 48th up to 63rd bits are never used to
// address RAM data so we can assume it's zero
a.li   (tmp, HPHP::safe_cast<int16_t>(
      (address & (ssize_t(UINT16_MAX) << 32)) >> 32));
a.sldi (tmp, tmp, 32);
a.addis(tmp, tmp, HPHP::safe_cast<int16_t>(
      (address & (ssize_t(UINT16_MAX) << 16)) >> 16));
a.addi (tmp, tmp, HPHP::safe_cast<int16_t>(
      address & ssize_t(UINT16_MAX)));
a.mtctr(tmp);
a.bcctr(bp.bo(), bp.bi(), 0);
```

Output code:

```
0x19e00000:  li      r3,1             
0x19e00004:  li      r4,2             
0x19e00008:  cmpd    r3,r4            
0x19e0000c:  blt     0x19e00008       
0x19e00010:  ble     0x19e00008       
0x19e00014:  bgt     0x19e00008       
0x19e00018:  bge     0x19e00008       
0x19e0001c:  beq     0x19e00008       
0x19e00020:  bne     0x19e00008       
0x19e00024:  bc      20,lt,0x19e00008 
0x19e00028:  bcl     20,lt,0x19e00008 
0x19e0002c:  bca     20,lt,0x1000     
0x19e00030:  bc      20,lt,0x19e00034 
0x19e00034:  bge     0x19e00008       
0x19e00038:  li      r11,0            
0x19e0003c:  rldicr  r11,r11,32,31    
0x19e00040:  addis   r11,r11,6624     
0x19e00044:  addi    r11,r11,56       
0x19e00048:  mtctr   r11              
0x19e0004c:  bctr                            
```
